tensorflow/python/ops/image_ops.py

```diff
@@ -427,36 +434,47 @@ def pad_to_bounding_box(image, offset_height, offset_width, target_height,
 
   Raises:
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
-      `target_*` arguments
+      `target_*` arguments, or either `offset_height` or `offset_width` is
+      negative.
   """
   image = ops.convert_to_tensor(image, name='image')
-  _Check3DImage(image, require_static=True)
+  _Check3DImage(image, require_static=False)
   height, width, depth = _ImageDimensions(image)
 
-  if target_width < width:
-    raise ValueError('target_width must be >= width')
-  if target_height < height:
-    raise ValueError('target_height must be >= height')
-
   after_padding_width = target_width - offset_width - width
   after_padding_height = target_height - offset_height - height
 
-  if after_padding_width < 0:
+  if not is_tensor(offset_height) and offset_height < 0:
+    raise ValueError('offset_height must be >= 0')
+  if not is_tensor(offset_width) and offset_width < 0:
+    raise ValueError('offset_width must be >= 0')
+
+  if (all(not is_tensor(i) for i in [target_width, width]) and
```

Comment:
for sets of 2, is_tensor(target_width) and is_tensor(width) is much easier to read